feat: add organization authentication notes for Connected Accounts flow (#1041)
auth0/docs-v2: auth4genai/snippets/how-tos/account-linking.mdx

If you're integrating with {props.connectionLabel || "any Identity Provider"}, but users in your app or agent can sign in using other methods (e.g., a username and password or another social provider), you'll need to link these identities into a single user account. Auth0 refers to this process as [Account Linking](https://auth0.com/docs/manage-users/user-accounts/user-account-linking).

Account Linking is separate from [Connected Accounts for Token Vault](https://auth0.com/docs/secure/tokens/token-vault/connected-accounts-for-token-vault):

- Account Linking merges multiple Auth0 identities into one user profile for sign-in.

> - Connected Accounts for Token Vault lets that signed-in user authorize an external provider so Auth0 can store the provider's tokens in Token Vault.

Reviewer: ```suggestion
- Connected Accounts for Token Vault allows that signed-in user to authorize an external provider so Auth0 can store the provider's tokens in Token Vault.
```

> If your application uses [Organizations](https://auth0.com/docs/manage-users/organizations), sign the user in to the correct organization before you start the Connected Accounts flow. Organizations define the session context; they do not create a shared external account for the organization.

Reviewer: ```suggestion
If your application uses [Organizations](https://auth0.com/docs/manage-users/organizations), authenticate the user with the target organization before initiating the Connected Accounts flow. Organizations define the session context; they do not create a shared external account for the organization.
```
